torch/_functorch/autograd_function.py

```diff
@@ -274,7 +274,17 @@ def validate_vmap_returns_tuple_of_two_elements(result):
 
 
 @custom_function_call.py_impl(TransformType.Vmap)
-def custom_function_call_vmap(interpreter, autograd_function, *operands):
+def custom_function_call_vmap(interpreter, autograd_function, *operands, **kwargs):
+    if any(
+        isinstance(val, torch.Tensor)
+        for val in torch.utils._pytree.tree_flatten(kwargs)[0]
+    ):
+        raise NotImplementedError(
+            f"Run vmap on autograd.Function with kwarg-only Tensor args. In the original "
+            f"definition of the op, please make your tensors not kwarg-only. "
+            f"Got: {kwargs}"
+        )
+
```

Comment:
check for kwarg-only tensor args added here.

Reply:
We have two routes, the autograd.Function route and the custom_op route. To check, is the only way to get here the autograd.Function route? If so we should remove the mention of "op" from the message: "Please do not pass kwarg-only Tensors to autograd.Function"